examples/instance/maas/instance.go

```diff
@@ -0,0 +1,282 @@
+	}()
+	<-isAcquired
+	params := url.Values{}
+	if _, err = acquiredNode.CallPost("start", params); err != nil {
+		return nil, err
+	}
+	hostname, err := acquiredNode.GetField("hostname")
+	if err != nil {
+		return nil, err
+	}
+	id := instance.ID(hostname)
+	machineDir, err := ioutil.TempDir(m.MaasfilesDir, "infrakit-")
+	if err != nil {
+		return nil, err
+	}
+	if err := ioutil.WriteFile(path.Join(machineDir, "MachineName"), []byte(hostname), 0755); err != nil {
+		return nil, err
+	}
+	tagData, err := types.AnyValue(spec.Tags)
+	if err != nil {
+		return nil, err
+	}
+	if err := ioutil.WriteFile(path.Join(machineDir, "tags"), tagData.Bytes(), 0666); err != nil {
+		return nil, err
+	}
+	if spec.LogicalID != nil {
+		if err := ioutil.WriteFile(path.Join(machineDir, "ip"), []byte(*spec.LogicalID), 0666); err != nil {
+			return nil, err
+		}
+	}
+	return &id, nil
+}
+
+// Label labels the instance
+func (m maasPlugin) Label(instance instance.ID, labels map[string]string) error {
+	machineDir := path.Join(m.MaasfilesDir, string(instance))
+	tagFile := path.Join(machineDir, "tags")
+	buff, err := ioutil.ReadFile(tagFile)
+	if err != nil {
+		return err
+	}
+
+	tags := map[string]string{}
+	err = types.AnyBytes(buff).Decode(&tags)
+	if err != nil {
+		return err
+	}
+
+	for k, v := range labels {
+		tags[k] = v
+	}
+
+	encoded, err := types.AnyValue(tags)
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(tagFile, encoded.Bytes(), 0666)
+}
+
+// Destroy terminates an existing instance.
+func (m maasPlugin) Destroy(id instance.ID) error {
+	fmt.Println("Destroying ", id)
+	nodeListing := m.MaasObj.GetSubObject("nodes")
+	listNodeObjects, err := nodeListing.CallGet("list", url.Values{})
+	if err != nil {
+		return err
+	}
+	listNodes, err := listNodeObjects.GetArray()
+	for _, nodeObj := range listNodes {
+		node, err := nodeObj.GetMAASObject()
+		hostname, err := node.GetField("hostname")
+		if hostname == string(id) {
+			if state, _ := node.GetField("substatus_name"); state == "Deploying" {
+				params := url.Values{}
+				if _, err = node.CallPost("abort_operation", params); err != nil {
+					return err
+				}
+			}
+			params := url.Values{}
+			if _, err = node.CallPost("release", params); err != nil {
+				return err
+			}
+		}
+	}
+	files, err := ioutil.ReadDir(m.MaasfilesDir)
+	if err != nil {
+		return err
+	}
+	for _, file := range files {
+		if !file.IsDir() {
+			continue
+		}
+		machineDir := path.Join(m.MaasfilesDir, file.Name())
+		hostname, err := ioutil.ReadFile(path.Join(machineDir, "MachineName"))
+		if err != nil {
+			if os.IsNotExist(err) {
+				continue
+			}
+			return err
+		} else if id == instance.ID(hostname) {
+			if err := os.RemoveAll(machineDir); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// DescribeInstances returns descriptions of all instances matching all of the provided tags.
+func (m maasPlugin) DescribeInstances(tags map[string]string) ([]instance.Description, error) {
```

Comment:
See comment above -- is there a way in MAAS to query by tags of the nodes?   Using files here is ok but we'd have to make sure these files are highly available.